packages/api/models/executions.js

```diff
@@ -118,11 +113,40 @@ class Execution extends Manager {
   * @returns {Promise<Object>} An execution record
    */
   async createExecutionFromSns(payload) {
-    const doc = this.generateDocFromPayload(payload);
-    doc.originalPayload = get(payload, 'payload');
-    doc.duration = (doc.timestamp - doc.createdAt) / 1000;
-    return this.create(doc);
+    return this.create(
+      this.buildNewExecutionRecordFromCumulusMessage(payload)
+    );
   }
-}
 
+  buildNewExecutionRecordFromCumulusMessage(cumulusMessage) {
```

Comment:
Needs a JSDoc, and maybe a test? If we change the Cumulus message in the future, tests for all these `buildXRecordFromCumulusMessage` functions should catch any problems that might cause.